libiop/bcs/merkle_tree.tcc

```diff
@@ -362,18 +399,23 @@ bool merkle_tree<FieldT, hash_digest_type>::validate_set_membership_proof(
     auto rand_it = proof.randomness_hashes.begin();
     auto aux_it = proof.auxiliary_hashes.begin();
 
-    typedef std::pair<std::size_t, hash_digest_type> pos_and_digest_t;
+    typedef std::pair<size_t, hash_digest_type> pos_and_digest_t;
     std::vector<pos_and_digest_t> S;
     S.reserve(positions.size());
 
     std::vector<hash_digest_type> leaf_hashes;
     if (this->make_zk_) {
         for (auto &leaf : leaf_contents)
         {
+            /* FIXME: This code is currently incorrect if the given list of positions is not
+               sorted or has duplicates. This could be fixed if both positions and leaf_contents
+               are sorted before the leaf hashes are calculated, which would require refactoring. */
```

Comment:
Is this a problem before this PR, or is it a new one? (My impression is that the way we add ZK is reasonably adding this limitation. Is the original okay when the positions are not sorted?)

Reply:
^ same as above. If possible, explicitly sort the positions and lift this limitation.